### ISSUE TITLE
ENH: An option on how density histograms handle outliers

### ISSUE THREAD

**Issue by bbbbbbbbba** [numpy/numpy#13604]
I am computing density histograms of a data set with some outliers that appear with small but non-negligible frequency (say around 10%). Including the outliers would skew the plot too much, so I exclude them using the `range` keyword argument. However, this causes `np.histogram` to ignore them completely, over-presenting the probability density everywhere else.

### Reproducing code example:

```python
import numpy as np
print(np.histogram([1.5, 3.5, 5.5, 7.5, 9.5, 100], bins = 10, range = (0, 10), density = True))

# (array([0. , 0.2, 0. , 0.2, 0. , 0.2, 0. , 0.2, 0. , 0.2]), array([ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9., 10.]))
```

Ideally, I want the density values to be 0.16666667 (1/6) instead of 0.2 (1/5).

### Current workaround:
Of course I can just choose not to use the `density = True` keyword argument, and manually divide by the number of samples and the bin width (easier with equal-width bins). However, this is cumbersome, especially when I am indirectly calling `np.histogram` through `matplotlib.pyplot.hist`. An option to directly compute the correct probability density would be welcome.

### Suggestion for implementation:
I think accounting for outliers when calculating the probability density might be the desired behavior in most cases where it matters. If this is the case, then maybe this behavior can be the default. In any case, it may be good to allow a "third value" (e.g. a specific string) for the `density` keyword argument to enable the non-default behavior.

**Comment by eric-wieser** [comment]
This is pretty surprising behavior to me, and I don't think the docs are nearly clear enough about it - so I'm not sure this was even deliberate.

**Comment by seberg** [comment]
True, the docs speak more about the non-density case I am sure. We would could go with a FutureWarning to change it (with the bonus of being noisy). I wonder if we should provide a way to get the old behaviour. I suppose someone may want it to disregard outliers, but then they probably can be expected to do that manually...

**Comment by seberg** [comment]
Sorry, the main reason why I suggested a new kwarg was actually, that we cannot change the `density` easily to flag for this. We already use `density=True`, and `density="something"` seems strange to get around a `FutureWarning`. So a new kwarg would solve the problem of allowing to opt-in and get no warning.

**Comment by bbbbbbbbba** [comment]
That makes sense. I was worried that a kwarg name like `ignore_outliers` might be somewhat cryptic, but we can probably already expect users of `np.histogram` to read the docs. Of course a `FutureWarning` should only be emitted if the dataset actually contains outliers.

**Comment by eric-wieser** [comment]
It's frustrating that we now got density wrong twice (the first time as `normed`). Ignoring the deprecation path, presumably the fix here is to make this change to `histogram`
```diff
-        return n/db/n.sum(), bin_edges
+        return n/db/len(a), bin_edges
```

I'm wondering how many people are relying on the current behavior - I'm tempted to call it an outright bug, meaning we could skip the warning entirely

**Comment by seberg** [comment]
I will raise it on the mailing list. But maybe we can also go the FutureWarning way, even if avoiding that warning is going to be hell.
